encoding/tsz/encoder.go

```diff
@@ -41,12 +41,13 @@ type encoder struct {
 	opts Options
 
 	// internal bookkeeping
-	t   time.Time       // current time
-	dt  time.Duration   // current time delta
-	vb  uint64          // current value
-	xor uint64          // current xor
-	ant m3db.Annotation // current annotation
-	tu  xtime.Unit      // current time unit
+	t         time.Time       // current time
+	dt        time.Duration   // current time delta
+	vb        uint64          // current value
+	xor       uint64          // current xor
+	ant       m3db.Annotation // current annotation
+	tu        xtime.Unit      // current time unit
+	tuChanged bool            // whether we have a new time unit
```

Comment:
Instead of adding a member field, I think returning a `bool` from `writeTimeUnit` is better whether it wrote it or not and then using that inside `writeTimeUnit` and passing it to the call to `writeDeltaOfDelta` as a parameter.  Thoughts?